### PR TITLE
Migrate benchmark to 1.23

### PR DESCRIPTION
**Pull request by eyalkraft**

### CODE REVIEW

File: compliance/cis_k8s/rules/cis_1_2_14/data.yaml [hunk 1]
```diff
@@ -1,36 +1,36 @@
 metadata:
-  id: 554a999e-c11a-5c1c-aa53-2badd123c122
-  name: Ensure that the admission control plugin NamespaceLifecycle is set
-    (Automated)
+  id: cf6ebe82-9367-5b8e-a117-c81ab11ba7d4
```
Comment (kfirpeled) [inline compliance/cis_k8s/rules/cis_1_2_14/data.yaml:2]: I don't mind now, but we should avoid changing ids of rules once we are out.
worth discussing it as we go forward about the id strategy in the tool

File: compliance/cis_k8s/rules/cis_1_2_14/data.yaml [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
-    None
+    None.
   default_value: |
-    By default, `NamespaceLifecycle` is set.
+    By default, `ServiceAccount` is set.
   references: |
     1. [https://kubernetes.io/docs/admin/kube-apiserver/](https://kubernetes.io/docs/admin/kube-apiserver/)
-    2. [https://kubernetes.io/docs/admin/admission-controllers/#namespacelifecycle](https://kubernetes.io/docs/admin/admission-controllers/#namespacelifecycle)
+    2. [https://kubernetes.io/docs/admin/admission-controllers/#serviceaccount](https://kubernetes.io/docs/admin/admission-controllers/#serviceaccount)
+    3. [https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/)
```
Comment (kfirpeled) [inline compliance/cis_k8s/rules/cis_1_2_14/data.yaml:33]: another note for editing rules guidelines tbd - bumping rule version number

File: compliance/cis_k8s/rules/cis_1_2_14/data.yaml (hunk 1 above)
Comment (eyalkraft) [reply compliance/cis_k8s/rules/cis_1_2_14/data.yaml:2]: maybe the right thing is to change the ids for all

Comment (eyalkraft) [reply compliance/cis_k8s/rules/cis_1_2_14/data.yaml]: did it with the help of @ofiriro3 and the script of @jeniawhite

Comment (kfirpeled) [reply compliance/cis_k8s/rules/cis_1_2_14/data.yaml]: i'll set up a discussion for this. not critical for this pr (srry)

**Comment by ofiriro3** [comment]
Did we create a task for Cloudbeat to remove PodSecurityPolicies from `kubeFetcher`?

**Comment by eyalkraft** [comment]
> Did we create a task for Cloudbeat to remove PodSecurityPolicies from kubeFetcher?

@ofiriro3 Nice catch!

- Opened https://github.com/elastic/cloudbeat/issues/75

File: compliance/cis_k8s/rules/cis_1_2_24/data.yaml [hunk 3]
```diff
@@ -1,39 +1,40 @@
...
   impact: |
     None
   default_value: |
-    By default, auditing is not enabled.
+    By default, `--service-account-lookup` argument is set to `true`.
   references: |
     1. [https://kubernetes.io/docs/admin/kube-apiserver/](https://kubernetes.io/docs/admin/kube-apiserver/)
-    2. [https://kubernetes.io/docs/concepts/cluster-administration/audit/](https://kubernetes.io/docs/concepts/cluster-administration/audit/)
-    3. [https://github.com/kubernetes/features/issues/22](https://github.com/kubernetes/features/issues/22)
+    2. [https://github.com/kubernetes/kubernetes/issues/24167](https://github.com/kubernetes/kubernetes/issues/24167)
```
Comment (ofiriro3) [inline compliance/cis_k8s/rules/cis_1_2_24/data.yaml:36]: Nit- redundant space

**Review by ofiriro3 — APPROVED** [review]

**Comment by ofiriro3** [comment]
Very good and tidy job! You rock!

**Pull request merged**